tests/commands/load/test_load_case_cmd.py

```diff
@@ -45,25 +45,104 @@ def test_load_case_from_ped(mock_app, institute_obj, case_obj):
     assert case_obj["genome_build"] == 37
 
 
-def test_load_case_from_yaml(mock_app, institute_obj, case_obj):
-    """Testing the scout load case command"""
+def test_load_case_case_keys(mock_app, institute_obj, case_obj, demo_case_keys):
+    """Testing loaded case contains expected keys when loaded using yaml file"""
 
     runner = mock_app.test_cli_runner()
-    assert runner
 
-    # remove case from real populated database using adapter
+    # GIVEN a database with no cases
     store.delete_case(case_id=case_obj["_id"])
     assert store.case_collection.find_one() is None
-    res = store.institute_collection.find({"_id": "cust000"})
-    assert sum(1 for i in res) == 1
 
-    # Make sure the scout config file is available
-    assert os.path.exists(load_path)
+    # GIVEN that the database contains cases HPO terms:
+    hpo_terms = [
+        {"_id": "HP:0001298", "hpo_id": "HP:0001298", "description": "Encephalopathy"},
+        {"_id": "HP:0001250", "hpo_id": "HP:0001250", "description": "Seizures"},
+    ]
+    store.hpo_term_collection.insert_many(hpo_terms)
 
-    # Test command to upload case using demo resources:
+    # WHEN a case is loaded using a yaml config file
     result = runner.invoke(cli, ["load", "case", load_path])
+
+    # THEN the command should be successful
     assert result.exit_code == 0
-    assert sum(1 for i in store.case_collection.find()) == 1
+
+    # AND the case should be saved in database
+    new_case = store.case_collection.find_one()
+    assert new_case
+
+    # WHEN case is loaded it should contain the expected fields:
+    key_value_template = "{key}:{bool_value}"  # for instance lims_id:True
+    key_value = None
+
+    for key in demo_case_keys:
+        expected_key_value = key_value_template.format(key=key, bool_value=True)
+
+        # Some fields might be boolean False values
+        if key in ["is_research", "research_requested", "rerun_requested", "is_migrated"]:
+            key_value = key_value_template.format(key=key, bool_value=new_case[key] is False)
+
+        # Some fields might not have values:
+        elif key in ["multiqc", "cnv_report", "coverage_qc_report", "gene_fusion_report_research"]:
+            key_value = key_value_template.format(key=key, bool_value=new_case[key] is None)
+
+        # Some other keys might be empty lists
+        elif key in ["dynamic_gene_list", "group"]:
+            key_value = key_value_template.format(key=key, bool_value=len(new_case[key]) == 0)
+
+        else:
+            key_value = key_value_template.format(key=key, bool_value=bool(new_case[key]))
```

Comment:
I think these checks are too tightly coupled with the example demo file.  For example if calling `bool()` on an empty string or zero will result in `False` breaking the test. This make the test implementation dependent on the config.

As I understand the purpose is to check that keys exists in the dict read from the DB. Would this be enough?


```
    # WHEN a case is loaded using a yaml config file
    result = runner.invoke(cli, ["load", "case", load_path])

    # THEN the command should be successful
    assert result.exit_code == 0

    # AND the case should be saved in database
    new_case = store.case_collection.find_one()
    assert new_case

    # WHEN case is loaded it should contain the expected fields:
    for key in demo_case_keys:
        assert key in new_case
```

Reply:
> This make the test implementation dependent on the config.

That's the intended behavior. The demo file is the only config file that reflects all the possible loadable values from the start for a case. It's fairly complete in providing an example for the demo case. If any of these tests start to fail it would means we have disrupted a function that should be working instead. 

> As I understand the purpose is to check that keys exists in the dict read from the DB. Would this be enough?

The demo case has a number of keys that exist but are None and it's fine. For some other keys it's different: if they exist but they are none after loading the demo case it means that something has gone wrong during the case loading. For instance the pedigree

This is the key after you load the case using master branch:
![image](https://user-images.githubusercontent.com/28093618/121011962-8462f900-c797-11eb-88db-a7ff4b3478c9.png)

This is the same key with None value when using the pydantic PR:
![image](https://user-images.githubusercontent.com/28093618/121012279-d9067400-c797-11eb-87fe-8343c8797195.png)

That's why I think it's important that we check that these keys not only exist, but have the expected values

Reply:
Yes, I agree that we should test for mandatory params and that they are set if needed (there are indeed some bugs in the pydantic-PR). But I think it could be done even better.

* Fail early. ie. as soon as we notice parsing is broken. Not after data has been written and read from DB
* Independent of underlying representation 

For example small changes in the config file might cause the tests to fail. Try setting 
`rank_model_version: ''`

Reply:
That's fine to check them earlier. Feel free to close this PR and introduce these changes in your Pydantic PR if you want! 👍🏻